src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -426,27 +248,49 @@ public virtual void Dispose()
 
                 _cts.Cancel();
                 _disposed = true;
-                CleanThreadLocalState();
-                ThreadLocalCleanup.ReleaseThreadLocalState -= CleanThreadLocalState;
-                _nativeMemoryCleaner.Dispose();
-                _nativeMemoryCleaner = null;
+                _idleTimer.Dispose();
+
+                while (_globalQueue.TryDequeue(out var result))
+                {
+                    result.Dispose();
```

Comment:
Do we need to check if in use?

Reply:
This is in the `Dispose`, caller is responsible for ensuring this.